packages/grid/test/basic.test.js

```diff
@@ -248,42 +248,42 @@ describe('basic features', () => {
     return column.renderer.getCalls().filter((call) => call.args[2].index === 0).length;
   }
 
-  it('should have rendered the first cell once', () => {
+  it('should render the first cell once during initialization', () => {
     expect(getFirstCellRenderCount()).to.equal(1);
   });
 
-  it('should re-render the cell when last row enters the viewport on resize', () => {
+  it('should re-render the first cell when last row enters the viewport on resize', () => {
     column.renderer.resetHistory();
     grid.size = 1;
-    expect(getFirstCellRenderCount()).to.equal(1);
+    expect(getFirstCellRenderCount()).to.equal(1); // once on size change
   });
 
-  it('should re-render the cell when last row leaves the viewport on resize', () => {
+  it('should re-render the first cell when last row leaves the viewport on resize', () => {
```

Comment:
Some of these tests relied on incorrect behavior where the grid failed to remove items beyond the new cache size.